distributed/worker_metrics.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+from prometheus_client import Counter, Gauge, Histogram, Summary
```

Comment:
TODO: `from distributed import prometheus`; there create stub classes when `prometheus_client` is not available

Reply:
I plan to evaluate switching to `opentelemetry` as brought up by @jacobtomlinson. From what I understand, this would allow us to introduce a hard dependency on the `opentelemetry-api` only (which contains a bunch of NoOpMetrics). I think this should solve your problem here.

Reply:
@crusaderky I assume the reason all these metrics are defined in one file (as opposed to being defined in the place they're used) is to have one place to deal with the potential import error?

I was a little surprised to see the metrics created in one place and used in another, when they're only used in exactly one place.

Reply:
No, it's because many metrics are contributed to by different modules.

For example for `dask_worker_event_loop_blocked`:
- The `disk-write-target` label is set by `worker_state_machine`
- The `disk-write-spill` label is set by `worker_memory`
- The `disk-read-execute` label is set by `worker`
- The `disk-read-get-data` label is set by `worker`

Besides that, it also felt nice to have everything in a nice, self-contained module and call the metrics through a `metrics.XXX` domain.